# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/MediaTransportHandler.cpp
# Commit: bac926c1886e
# Full Hash: bac926c1886edd2bb5cdb16c66ae24aa6e23d341
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-14 11:16:09
# Description:
#   Bug 1577719: Init r_log in the c'tor, to simplify things. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45463
# ==============================================================================

diff -r 9087e4844e75 -r bac926c1886e media/webrtc/signaling/src/peerconnection/MediaTransportHandler.cpp
--- a/media/webrtc/signaling/src/peerconnection/MediaTransportHandler.cpp	Fri Sep 13 22:12:05 2019 +0000
+++ b/media/webrtc/signaling/src/peerconnection/MediaTransportHandler.cpp	Wed Sep 11 02:29:15 2019 +0000
@@ -233,6 +233,9 @@
   if (!mStsThread) {
     MOZ_CRASH();
   }
+
+  RLogConnector::CreateInstance();
+
   CSFLogDebug(LOGTAG, "%s done", __func__);
 
   // We do not set up mDNSService here, because we are not running on main (we
@@ -1080,9 +1083,6 @@
 }
 
 void MediaTransportHandlerSTS::EnterPrivateMode() {
-  // Do this from calling thread, because that's what we do in CreateIceCtx...
-  RLogConnector::CreateInstance();
-
   if (!mStsThread->IsOnCurrentThread()) {
     mStsThread->Dispatch(
         WrapRunnable(RefPtr<MediaTransportHandlerSTS>(this),
@@ -1095,9 +1095,6 @@
 }
 
 void MediaTransportHandlerSTS::ExitPrivateMode() {
-  // Do this from calling thread, because that's what we do in CreateIceCtx...
-  RLogConnector::CreateInstance();
-
   if (!mStsThread->IsOnCurrentThread()) {
     mStsThread->Dispatch(
         WrapRunnable(RefPtr<MediaTransportHandlerSTS>(this),
